crates/composefs/src/repository.rs

```diff
@@ -335,6 +335,12 @@ impl<ObjectID: FsVerityHashValue> Repository<ObjectID> {
         Ok(object_id)
     }
 
+    pub fn has_named_stream(&self, name: &str) -> bool {
+        let stream_path = format!("streams/refs/{}", name);
+
+        readlinkat(&self.repository, &stream_path, []).is_ok()
```

Comment:
I don't like "swallowing" errors like this, I'd say call `stat` instead and require it's `S_IFLNK`